### PR TITLE
Feature/jsonrpc basic impl

### PR DESCRIPTION
**Pull request by jm-clius**
This is an incremental PR towards #299.

It adds a basic polling call to the API, whereby a client can receive messages on subscribed topics by polling an in-memory cache on the server. The cache clears after every call, so that only new messages are returned by each new call.

Still outstanding:
- ability to limit the number of cached topics and messages (NB to avoid memory leaks)
- ability to specify the number of messages to read
- priming the cache with topics that the node is already subscribed to

### CODE REVIEW

File: tests/v2/test_jsonrpc_waku.nim [hunk 1]
```diff
@@ -106,6 +106,77 @@ procSuite "Waku v2 JSON-RPC API":
...
+      server = newRpcHttpServer([ta])
+    
+    # Let's connect to node 3 via the API
+    installRelayApiHandlers(node3, server)
+    server.start()
+
+    let client = newRpcHttpClient()
+    await client.connect("127.0.0.1", rpcPort)
+
+    echo "Now try to subscribe using API"
```
Comment (oskarth) [inline tests/v2/test_jsonrpc_waku.nim:148]: Maybe DEBUG?

File: tests/v2/test_jsonrpc_waku.nim [hunk 2]
```diff
@@ -106,6 +106,77 @@ procSuite "Waku v2 JSON-RPC API":
     server.stop()
     server.close()
     waitfor node.stop()
+  
+  asyncTest "relay_polling":
```
Comment (oskarth) [inline tests/v2/test_jsonrpc_waku.nim:110]: Can we have a more descriptive name of what this is testing?

File: waku/v2/node/jsonrpc/relay_api.nim [hunk 3]
```diff
@@ -1,12 +1,32 @@
...
   ../wakunode2,
   ./jsonrpc_types, ./jsonrpc_utils
 
+const futTimeout = 5.seconds
+
 proc installRelayApiHandlers*(node: WakuNode, rpcsrv: RpcServer) =
-  const futTimeout = 5.seconds
+  ## Create a per-topic message cache
+  var
+    topicCache = initTable[string, seq[WakuMessage]]()
```
Comment (oskarth) [inline waku/v2/node/jsonrpc/relay_api.nim:17]: Hm, does this state belong in JSONRPC code? I suppose that's the only place it is currently used. Seems a bit odd to me though, with this local state? Not sure.

**Review by oskarth — APPROVED** [review]
Generally looks good, some comments and questions in line

File: tests/v2/test_jsonrpc_waku.nim (hunk 1 above)
Comment (jm-clius) [reply tests/v2/test_jsonrpc_waku.nim:148]: Ah, missed this. Thanks. Should certainly be removed.

File: tests/v2/test_jsonrpc_waku.nim (hunk 2 above)
Comment (jm-clius) [reply tests/v2/test_jsonrpc_waku.nim:110]: Will do!

File: waku/v2/node/jsonrpc/relay_api.nim (hunk 3 above)
Comment (jm-clius) [reply waku/v2/node/jsonrpc/relay_api.nim:17]: Good question. After considering a couple of options I settled on this being the _safest_ place for the state, where it is as short- or long-lived as the RPC handlers and will be garbage collected with everything else. Still, I will definitely consider a better (more global) place for this, though asynchronous read/write access to the table may become tricky.

**Pull request merged**